bundles/org.openhab.ui.dashboard/templates/index.html

```diff
@@ -14,6 +14,25 @@
 
 <script type="text/javascript" src="js/jquery.min.js"></script>
 <script type="text/javascript" src="js/geolocation.js"></script>
+<script>
+$(function(){
+    var url = new URL(window.location.href);
+    console.log( "url: " + url.host );
+    $('div[class="link"]').each(function(index,item){
+        if ($(item).attr('onclick').indexOf('{HOST}') >= 0) {
```

Comment:
I would say that the whole concept of hacking div into a clickable button is quite wrong.
Right now it looks like this:
```html
                <div class="link" onclick="location.href = '{HOST}/flowsbuilder/index.html'">
                    <div class="body" style="background-image: url('{HOST}/flowsbuilder/tile.png')">
                         <div class="overlay icon-none"></div>
                    </div>
                    <div class="footer">
                        <p>Flows Builder</p>
                    </div>
                </div>
```
My proposal would be to 
1. rename `div[class="link"]` into simple `a.link`
1. Utilize anchor's `href` property

so it looks like this:
```html
                <a class="link" href="{HOST}/flowsbuilder/index.html">
                    <div class="body" style="background-image: url('{HOST}/flowsbuilder/tile.png')">
                         <div class="overlay icon-none"></div>
                    </div>
                    <div class="footer">
                        <p>Flows Builder</p>
                    </div>
                </a>
```

---

Now in the JS part we just need to collect all those links and replace {HOST} with our origin.
No need to check `indexOf`.

```js
<script type="text/javascript">
$(document).ready(function() {
    $('a.link').each(function(index) {
        var href = $(this)
            .attr('href')
            .replace(/{HOST}/g, window.location.host)
            .replace(/{HOSTNAME}/g, window.location.hostname);
        $(this).attr('href', href);
    });
});
</script>
```

Reply:
I also spotted that, that's why I've mentioned you 🤛